src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/KcStreamingIngestClient.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.snowflake.kafka.connector.internal.ingestsdk;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.internal.LoggerHandler;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+/** This is a wrapper to help define a contract with the streaming ingest clients */
+public class KcStreamingIngestClient {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT";
+  private LoggerHandler LOGGER = new LoggerHandler(this.getClass().getName());
+
+  private final SnowflakeStreamingIngestClient client;
+
+  /**
+   * Builds the client's name based on the kc instance and expected client id
+   *
+   * @param kcInstanceId the kafka connector instance id
+   * @param clientId the client id
+   * @return the client's name as 'KC_CLIENT_kcInstanceId_clientId'
+   */
+  public static String buildStreamingIngestClientName(String kcInstanceId, int clientId) {
+    return Utils.formatString("{}_{}_{}", STREAMING_CLIENT_PREFIX_NAME, kcInstanceId, clientId);
+  }
+
+  // TESTING ONLY - inject the client
+  @VisibleForTesting
+  public KcStreamingIngestClient(SnowflakeStreamingIngestClient client) {
+    this.client = client;
+  }
+
+  /**
+   * Creates a streaming client from the given properties and requested name. Validates the
+   * requested client is not null and has the correct name Any exceptions will be passed up, a
+   * sfexception will be converted to a connectexception
+   *
+   * @param streamingClientProps the properties for the client
+   * @param clientName the client name to uniquely identify the client
+   */
+  protected KcStreamingIngestClient(Properties streamingClientProps, String clientName) {
+    try {
+      LOGGER.info("Creating Streaming Client: {}", clientName);
+      this.client =
+          SnowflakeStreamingIngestClientFactory.builder(clientName)
+              .setProperties(streamingClientProps)
+              .build();
+
+      assert this.client != null; // client is final, so never need to do another null check
+      assert this.client.getName().equals(clientName);
```

Comment:
What's the point of this check?

Reply:
Same as the null check motivation - a sanity check, and throwing the exception early if the streaming client behavior ever changes

Reply:
resolving this, similar to this comment https://github.com/snowflakedb/snowflake-kafka-connector/pull/528#discussion_r1095087551